syntax = "proto3";

package fixbug;
option cc_generic_services = true;

message ResultCode 
{
    int32 err_code = 1;
    bytes err_msg = 2;
}

message LoginRequest
{
    bytes name  = 1;
    bytes pwd = 2;
}

message LoginResponse
{
    ResultCode rscode = 1;
    bool success = 2;
}
message RegisterRequest
{
    uint32 id = 1;
    bytes name  = 2;
    bytes pwd = 3;
}

message RegisterResponse
{
    ResultCode rscode = 1;
    bool success = 2;
}

service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}

message GetFriendListRequest
{
    uint32 id = 1;
}

message GetFriendListResponse
{
    ResultCode rscode = 1;
    repeated bytes friends = 2;
}


service FriendServiceRpc
{
    rpc GetFriendList(GetFriendListRequest) returns(GetFriendListResponse);
}
